doc/api/http.md

```diff
@@ -917,6 +917,65 @@ Example:
 var contentType = response.getHeader('content-type');
 ```
 
+### response.getHeaderNames()
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Array}
+
+Returns an array containing the names of the current outgoing headers. All
+header names are lowercase.
+
+Example:
+
+```js
+response.setHeader('Foo', 'bar');
+response.setHeader('Set-Cookie', ['foo=bar', 'bar=baz']);
+
+var headerNames = response.getHeaderNames();
+// headerNames === ['foo', 'set-cookie']
+```
+
+### response.getHeaders()
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Object}
+
+Returns a shallow copy of the current outgoing headers. The keys of the
```

Comment:
Is this true? Below, you say composite values, Arrays, are also shallow copied. When you recursively shallow copy, you get a deep copy. Why not just say its a copy, and may be safely modified by the user? You have an eye on performance, so I asssume it wouldn't help, but would freezing instead of copying be faster or slower?